test/planner/plan_util_test.cpp

```diff
@@ -0,0 +1,160 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// plan_util_test.cpp
+//
+// Identification: test/planner/plan_util_test.cpp
+//
+// Copyright (c) 2015-18, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "common/harness.h"
+
+#include "catalog/catalog.h"
+#include "catalog/database_catalog.h"
+#include "catalog/index_catalog.h"
+#include "catalog/table_catalog.h"
+#include "common/statement.h"
+#include "concurrency/transaction_manager_factory.h"
+#include "executor/testing_executor_util.h"
+#include "parser/postgresparser.h"
+#include "storage/data_table.h"
+
+#include "planner/plan_util.h"
+
+namespace peloton {
+namespace test {
+
+#define TEST_DB_NAME "test_db"
+
+class PlanUtilTests : public PelotonTest {};
+
+TEST_F(PlanUtilTests, GetAffectedIndexesTest) {
+  auto catalog = catalog::Catalog::GetInstance();
+  catalog->Bootstrap();
+
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+
+  catalog->CreateDatabase(TEST_DB_NAME, txn);
+  auto db = catalog->GetDatabaseWithName(TEST_DB_NAME, txn);
+  // Insert a table first
+  auto id_column = catalog::Column(
```

Comment:
We should change how we do this and creating indexes.

Reply:
You mean using the approach in old_optimizer_test? I looked at both the approaches and this looked better to me (using methods instead of manually typing queries to create tables, indexes in). And I couldn't find any downsides of this approach.

Reply:
Oh if you can get it work without adding the extra TryBindDatabaseName function then I think it's fine. Then like you said, it would would be better to factor this db, table, index creation under a new set of `TestingExecutorUtil`, and ideally update the DropTests to use this as well.

Reply:
I have already done so. :)

Reply:
Cool! Can you factor this out to TestingExecutorUtil as a new utility function as well? Then other tests can also use this. Then I think it’s only the compilation issue left.